components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1382,6 +1515,18 @@ where
                 self.register_raft_gc_log_tick();
                 self.register_check_leader_lease_tick();
             }
+
+            if let Some(ForceLeaderState::ForceLeader) = self.fsm.peer.force_leader {
+                if r != StateRole::Leader {
+                    // for some reason, it's not leader anymore
+                    info!(
```

Comment:
Don't understand, can you elaborate on this?

Reply:
Oh, what I'm asking is how to retry if it fails to collect enough votes in an election timeout.

Reply:
In practice, it's quite normal to complete campaign in 2 ~ 3 rounds. The failure rate may be high.

Reply:
Okay. It will issue a new round of request-vote after the election timeout. And `check_force_leader` will pass when receiving enough votes. Will add a test for it.